src/systemctl/systemctl-edit.c

```diff
@@ -371,60 +348,33 @@ int verb_edit(int argc, char *argv[], void *userdata) {
                         return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Cannot edit %s: unit is masked.", *tmp);
         }
 
-        r = find_paths_to_edit(bus, names, &edit_files);
+        r = find_paths_to_edit(bus, &context, names);
         if (r < 0)
                 return r;
-        if (!edit_files)
+        if (!context.files)
```

Comment:
Please check by `context.n_files == 0`. The buffer may be allocated, but may not be used.